### ISSUE TITLE
Migration to mitaka build from liberty build fails on keystone

### ISSUE THREAD

**Issue by kamidzi** [bloomberg/chef-bcpc#1200]
```
* bash[wait-for-keystone-to-become-operational] action run

    ================================================================================
    Error executing action `run` on resource 'bash[wait-for-keystone-to-become-operational]'
    ================================================================================
```

When switching `openstack_release` to `mitaka` and re-cheffing.

**Comment by erhudy** [comment]
Keystone appears to be failing to start up because of a SQL error, which indicates that the schema transformation isn't running as expected. Should be relatively easy to fix.

**Comment by erhudy** [comment]
If you run Chef again, it should get past that broken point. The underlying issue is that the check to determine whether it's an upgrade (https://github.com/bloomberg/chef-bcpc/pull/1151/files#diff-ac80043a573ee417b2b7494ac11ae514R42) happens at compile time rather than runtime, and the first time you run Chef to start the upgrade the updated OpenStack package repository hasn't been introduced, so the check fails, the upgrade flag file isn't created, the database schema update doesn't happen, and Keystone fails to start. I will push a fix commit onto the Mitaka branch.

**Comment by erhudy** [comment]
Fix pushed.

**Comment by kamidzi** [comment]
Closed via bb2a29bfb8952d6bcdc9db6aaeb9caa62bc626af
